src/ui/widgets/Label/label.tsx

```diff
@@ -41,61 +42,57 @@ export const LabelComponent = (
   props: InferWidgetProps<typeof LabelProps>
 ): JSX.Element => {
   // Default labels to transparent.
-  const editedProps = {
-    ...props,
-    transparent: props.transparent ?? true
-  };
-  const style: CSSProperties = commonCss(editedProps);
   const {
+    transparent = true,
+    foregroundColor = diamondTheme.palette.primary.contrastText,
     textAlign = "left",
     textAlignV = "top",
     text = "",
-    rotationAngle,
+    rotationStep = 0,
     wrapWords
   } = props;
-  const className = props.className ?? `Label ${classes.Label}`;
+  const backgroundColor = transparent
+    ? "transparent"
+    : (props.backgroundColor?.toString() ?? diamondTheme.palette.primary.main);
+  const font = props.font?.css() ?? diamondTheme.typography;
+
   // Since display is "flex", use "flex-start" and "flex-end" to align
   // the content.
   let alignment = "center";
   if (textAlign === "left") {
     alignment = "flex-start";
-    if (wrapWords) {
-      style["textAlign"] = "left";
-    }
   } else if (textAlign === "right") {
     alignment = "flex-end";
-    if (wrapWords) {
-      style["textAlign"] = "right";
-    }
-  } else {
-    if (wrapWords) {
-      style["textAlign"] = "center";
-    }
   }
-  style["justifyContent"] = alignment;
-  style["cursor"] = "default";
+
   let alignmentV = "center";
   if (textAlignV === "top") {
     alignmentV = "flex-start";
   } else if (textAlignV === "bottom") {
     alignmentV = "flex-end";
   }
-  style["alignItems"] = alignmentV;
-  let transform = undefined;
-  if (rotationAngle) {
-    transform = `rotate(${rotationAngle}deg)`;
-  }
-
-  if (wrapWords) {
-    style["wordBreak"] = "break-word";
-    style["whiteSpace"] = "break-spaces";
-  }
 
   // Simple component to display text - defaults to black text and dark grey background
   return (
-    <div className={className} style={style}>
```

Comment:
I'd suggest either:

- Keeping using this CSS class. It already contains properties like `width: "100%"`, `height: "100%"` and display: "flex" which saves us duplicating those lines a little below this.
- Don't use the CSS class, but delete the `label.module.css `file as it is now obsolete.